src/main/java/com/questhelper/quests/beneathcursedsands/BeneathCursedSands.java

```diff
@@ -0,0 +1,591 @@
+
+		placeEmblemsInUrns = new ObjectStep(this, NullObjectID.NULL_20391, new WorldPoint(3391, 9251, 0), "Inspect the north-western plaque and read it. The solution to the riddle is different for everyone. Please check the wiki guide to solve this.", true);
+		((ObjectStep) placeEmblemsInUrns).addAlternateObjects(NullObjectID.NULL_44587, NullObjectID.NULL_44588, NullObjectID.NULL_44589, NullObjectID.NULL_44590);
+		((ObjectStep) placeEmblemsInUrns).setHideWorldArrow(true);
+
+		pullRiddleLever = new ObjectStep(this, NullObjectID.NULL_20288, new WorldPoint(3390, 9247, 0), "Pull the lever to the south-west.");
+
+		enterTombDoor = new ObjectStep(this, ObjectID.DOOR_43962, new WorldPoint(3389, 9248, 0), "Enter the tomb.");
+
+		talkToSpirit = new NpcStep(this, NpcID.SPIRIT, new WorldPoint(3377, 9248, 0), "Speak to the Spirit.");
+
+		takeRustyKey = new ObjectStep(this, NullObjectID.NULL_44591, new WorldPoint(3368, 9248, 0), "Take the rusty key from the urn in the back of the room.");
+
+		// The Champion of Scabaras
+		leaveTombDoor = new ObjectStep(this, ObjectID.DOOR_43962, new WorldPoint(3389, 9248, 0), "Leave the tomb.");
+
+		leaveRiddleDoor = new ObjectStep(this, ObjectID.DOOR_43961, new WorldPoint(3405, 9248, 0), "Leave the riddle area.");
+
+		climbUpstairs = new ObjectStep(this, ObjectID.STAIRS_43956, "Go back upstairs.");
+		((ObjectStep) climbUpstairs).addAlternateObjects(ObjectID.STAIRS_43957);
+		((DetailedQuestStep) climbUpstairs).setLinePoints(Arrays.asList(new WorldPoint(3408, 9248, 0), new WorldPoint(3411, 9244, 0), new WorldPoint(3415, 9248, 0), new WorldPoint(3431, 9248, 0), new WorldPoint(3431, 9239, 0), new WorldPoint(3438, 9239, 0), new WorldPoint(3438, 9243, 0)));
+
+		unlockBossDoor = new ObjectStep(this, ObjectID.DOOR_43960, "Prepare to fight the Champion of Scabaras, and unlock the door to start the fight.", combatGearChampionOfScabaras, food, antipoison, staminaPotions, prayerPotions, rustyKey);
+		unlockBossDoor.addIcon(ItemID.RUSTY_KEY);
+
+		openBossDoor = new ObjectStep(this, ObjectID.DOOR_43960, "Prepare to fight the Champion of Scabaras, and open the door to start the fight.", combatGearChampionOfScabaras, food, antipoison, staminaPotions, prayerPotions);
+
+		fightChampionOfScabaras = new NpcStep(this, NpcID.CHAMPION_OF_SCABARAS_11483, "Fight the Champion of Scabaras.", combatGearChampionOfScabaras, food, antipoison, staminaPotions, prayerPotions);
+		fightChampionOfScabaras.addText("Protect from Magic, and keep 4+ tiles distance at all times.");
+		fightChampionOfScabaras.addText("He is weak to ranged attacks, so bring your best ranged gear.");
+		fightScarabSwarm = new NpcStep(this, NpcID.SCARAB_SWARM_11484, "Defeat the Swarm as soon as possible, to prevent shadow flames from hindering your movement.");
+		destroyShadowRift = new NpcStep(this, NpcID.SHADOW_RIFT, "Destroy the Shadow Rift as soon as possible, to prevent 35+ damage from occurring once it explodes.");
```

Comment:
These should be textual information in `fightChampionOfScabaras` imo, to avoid hand-holding in the fight in line with other difficult quest bosses with special moves (e.g. Fragment of Seren).

Reply:
Yeah, I thought about that, but somehow it felt overly crowded both in the Sidebar panel as well as in the top-left Quest Helper corner. I'll let someone else make a call on how to properly adjust this in-line with other quests/the QH style.